ecs-cli/modules/utils/compose/convert_task_definition.go

```diff
@@ -99,24 +124,20 @@ func ConvertToTaskDefinition(taskDefinitionName string, context *project.Context
 		containerDef := &ecs.ContainerDefinition{
 			Name: aws.String(name),
 		}
-		if err := convertToContainerDef(context, serviceConfig, volumes, containerDef); err != nil {
-			return nil, err
+
+		// Check if essential is specified for the container
+		ecsContainerDef := &ContainerDef{}
+		if cd, ok := ecsContainerDefs[name]; ok {
```

Comment:
Fun Golang thing that I think will let you reduce this all to 1 line. If you only capture the first return from accessing a map, and the key does not exist in the map, then you get the default value of the value type. This means that:

`ecsContainerDef := ecsContainerDefs[name]`

Should give you an empty struct if name is not a valid key.

Reply:
Hm, interesting suggestion. I think I prefer this idiom of checking whether an element is in a map though -- seems more clear to me!